pkg/factory/risingwave_object_factory.go

```diff
@@ -269,6 +269,20 @@ func (f *RisingWaveObjectFactory) newService(component string, serviceType corev
 	}
 }
 
+// getMallocEnv returns the environment variables for triggering heap dump without interval-triggered-dumps.
+func (f *RisingWaveObjectFactory) getMallocEnv() []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name:  envs.RWMallocConf,
+			Value: "prof:true,lg_prof_interval=-1,lg_prof_sample=20,prof_prefix:/var/mnt/coredump/heap",
+		},
+		{
+			Name:  envs.RWHeapProfilingDir,
+			Value: "/var/mnt/coredump/",
```

Comment:
Does this folder exist in both on-prem & cloud env?

Reply:
No, and that's why I believe this is not suitable to have in the operator.

Reply:
@fuyufjh do you know what happens if the folder does not exist? Will it throw an error or just not create heap files?
@arkbriar Do you have another suggestion where to put this, if not in the operator?

Reply:
I think we should put it in the cloud control plane.